pkg/vm/engine/memoryengine/table_reader.go

```diff
@@ -48,6 +48,9 @@ func (t *Table) NewReader(
 	readers []engine.Reader,
 	err error,
 ) {
+	if len(shardIDs) > 0 && len(shardIDs[0]) == 0 { // Compatible with cn's logic
```

Comment:
add a comments that if it matches the condition, it's mem table.
or add a function like
`func IsMemTable(shardIds) {
 return
}`
to do this thing.

Reply:
chuli le